tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -2564,15 +2564,34 @@ class Subgraph {
     TF_LITE_ENSURE_STATUS(CheckTensorNonDynamicAllocation(
         logging_context, output_tensor, output_tensor_id, node_index));
 
+    std::vector<int32_t> input_shape;
+    input_shape.assign(&input_tensor.dims->data[0],
+                       &input_tensor.dims->data[0] + input_tensor.dims->size);
     if (detect_supported_op) {
       TF_LITE_ENSURE_STATUS(CheckWebNNOpSupport(builder, "softmax"));
     } else {
       TF_LITE_ENSURE(logging_context,
                      webnn_operands.at(input_tensor_id).as<bool>());
-      webnn_operands.insert(
-          std::make_pair(output_tensor_id,
-                         builder.call<emscripten::val>(
-                             "softmax", webnn_operands.at(input_tensor_id))));
+      emscripten::val output = emscripten::val::object();
+      // WebNN Softmax only support 2d input shape, reshape input to 2d.
+      if (input_shape.size() != 2) {
+        emscripten::val new_shape = emscripten::val::array();
+        new_shape.call<void>("push", emscripten::val::null());
```

Comment:
Dwayne proposed to remove the support of `null` in newShape. https://github.com/webmachinelearning/webnn/issues/388

You may want to add a TODO comment here, once it is in spec and implementation. We should compute the dimension size by the delegate.

Reply:
Thanks for the reminder, TODO added.